chef/cookbooks/nova/recipes/config.rb

```diff
@@ -190,31 +190,19 @@
   cinder_insecure = cinder_server[:cinder][:api][:protocol] == 'https' && cinder_server[:cinder][:ssl][:insecure]
 
   if node.roles.include? "nova-multi-compute-kvm"
-    include_ceph_recipe = false
-
+    rbd_enabled = false
     cinder_server[:cinder][:volumes].each do |volume|
-      next if volume['backend_driver'] != "rbd"
-
-      # if include_ceph_recipe is already true, avoid re-entering the if (and executing a slow search)
-      if volume['rbd']['use_crowbar'] && !include_ceph_recipe
-        ceph_env_filter = " AND ceph_config_environment:ceph-config-default"
-        ceph_servers = search(:node, "roles:ceph-osd#{ceph_env_filter}") || []
-        if ceph_servers.length > 0
-          include_ceph_recipe = true
-        end
-      end
-    end
-
-    if include_ceph_recipe
-      include_recipe('ceph::nova')
-      ceph_user = node['ceph']['nova-user']
-      ceph_uuid = node['ceph']['nova-uuid']
+      rbd_enabled = true if volume['backend_driver'] == "rbd"
     end
   end
 else
   cinder_insecure = false
 end
 
+if rbd_enabled
+  include_recipe "nova::ceph"
+end
```

Comment:
You need to move the two lines at https://github.com/mjura/barclamp-nova/blob/multi_ceph_backends/chef/cookbooks/nova/recipes/config.rb#L184 after this, don't you?